packages/legend-query/src/components/__tests__/QueryBuilder_FetchStructure.test.tsx

```diff
@@ -148,6 +149,34 @@ test(
     expect(lastNameProperty).toBe(_personClass.getProperty('lastName'));
     expect(queryBuilderState.resultSetModifierState.limit).toBeUndefined();
 
+    // simpleProjection with subType
```

Comment:
@MauricioUyaguari this 200-line compound test is freaking me out. Should we split this up?

Reply:
Yes. @YannanGao-gs. Lets split these out. Each time we initialize queryBuilderState we should have a new test.